engine/src/main/java/org/camunda/bpm/engine/impl/dmn/entity/repository/DecisionDefinitionQueryImpl.java

```diff
@@ -291,6 +297,19 @@ public void checkQueryOk() {
     }
   }
 
+  private boolean checkDmnEnabled() {
+      if (dmnEnabled == null) {
+          ProcessEngineConfigurationImpl configuration = Context.getProcessEngineConfiguration();
+          if (configuration != null) {
+              dmnEnabled = Boolean.valueOf(configuration.isDmnEnabled());
+          } else {
+              dmnEnabled = Boolean.TRUE;
+          }
+          
+      }
+      return dmnEnabled.booleanValue();
```

Comment:
Why do we need the `dmnEnabled` flag? A new instance of the `DecisionDefinitionQueryImpl` class is created when the appropriate `RepositoryService` method is called. Wouldn't the following be enough:

```suggestion
      return Context.getCommandContext()
          .getProcessEngineConfiguration()
          .isDmnEnabled();
```